httpclient5/src/main/java/org/apache/hc/client5/http/SystemDefaultDnsResolver.java

```diff
@@ -40,7 +40,23 @@ public class SystemDefaultDnsResolver implements DnsResolver {
 
     @Override
     public InetAddress[] resolve(final String host) throws UnknownHostException {
-        return InetAddress.getAllByName(host);
+        try {
+            // Try resolving using the default resolver
+            return InetAddress.getAllByName(host);
+        } catch (final UnknownHostException e) {
+            // If default resolver fails, try stripping the IPv6 zone ID and resolving again
+            String strippedHost = host;
+            if (host.charAt(0) == '[') {
+                final int i = host.lastIndexOf('%');
+                if (i != -1) {
+                    strippedHost = host.substring(0, i) + "]";
+                }
+            }
+            if (!strippedHost.equals(host)) {
```

Comment:
@arturobernalg The whole point of inlining the method was to avoid this extra comparison! Please use a boolean variable instead or set `strippedHost` by default.

Reply:
@arturobernalg In fact if you use null to signify a case when the original host is unchanged (not stripped) and no second DNS is not needed you can put this code into a separate method as before, if you like it better that way.

Reply:
@arturobernalg How about this?
```
String strippedHost = null;
if (host.charAt(0) == '[') {
  final int i = host.lastIndexOf('%');
  if (i != -1) {
    strippedHost = host.substring(0, i) + "]";
  }
}
if (strippedHost != null) {
  return InetAddress.getAllByName(strippedHost);
}
throw e;
```

Reply:
@ok2c , there is no significant difference between the two code snippets you provided. Both snippets attempt to strip the IPv6 zone ID from the host name if it is present and retry the resolution if the default resolver fails.
But I any case, I made the change

Reply:
@arturobernalg There is a biiiig difference in terms of performance between `if (strippedHost != null)` and `if (!strippedHost.equals(host))`